uPortal-groups/uPortal-groups-core/src/main/java/org/apereo/portal/groups/RDBMEntityGroupStore.java

```diff
@@ -471,30 +484,19 @@ public Iterator findParentGroups(IGroupMember gm) throws GroupsException {
     public Iterator findEntitiesForGroup(IEntityGroup group) throws GroupsException {
         Collection entities = new ArrayList();
         Connection conn = null;
+        PreparedStatement ps = null;
         String groupID = group.getLocalKey();
         Class cls = group.getLeafType();
 
         try {
             conn = RDBMServices.getConnection();
-            Statement stmnt = conn.createStatement();
+            ps = conn.prepareStatement(SEARCH_ENTITIES_FOR_GROUP);
```

Comment:
I'm not as familiar with connection behavior here.
Does this mean the query will be re-prepared each time it is called?
Or is this clever and it avoid preparing it again if it already exists?

Reply:
I think you're correct that it will re-prepare the statement each time. This is how the same sort of code is used in several other instances in this file though, so this is the established pattern. It might be worth a ticket to perhaps revisit those and make it so the statement is only prepared once, if that would still work?

Reply:
> This is how the same sort of code is used in several other instances in this file though, so this is the established pattern

Right

> I think you're correct that it will re-prepare the statement each time
> It might be worth a ticket to perhaps revisit those and make it so the statement is only prepared once, if that would still work?

Absolutely, it doesn't need to be a part of this PR, I was mainly curious. :heavy_check_mark: 
If there's time a follow up task/PR to do some optimizations would be welcome. :slightly_smiling_face: